shopinvader/tests/test_cart.py

```diff
@@ -440,6 +440,32 @@ def test_cart_misc_data_update(self):
         cart = self.cart
         self.assertEqual(cart.client_order_ref, "#SpecialPurchaseDude!")
 
+    def test_cart_get_from_any_partner(self):
+        anon_cart = self.env.ref("shopinvader.sale_order_1")
+        with self.work_on_services(
+            partner=self.partner,
+            shopinvader_session={"cart_id": anon_cart.id},
+        ) as work:
+            self.service = work.component(usage="cart")
+            cart = self.service.dispatch("search")["data"]
+
+        self.assertEqual(anon_cart.id, cart["id"])
+        self.assertNotEqual(anon_cart.partner_id, self.partner)
+
+    def test_cart_get_restrict_cart_to_partner(self):
+        self.backend.restrict_cart_to_partner = True
+        anon_cart = self.env.ref("shopinvader.sale_order_1")
+        with self.work_on_services(
+            partner=self.partner,
+            shopinvader_session={"cart_id": anon_cart.id},
+        ) as work:
+            self.service = work.component(usage="cart")
+            cart = self.service.dispatch("search")["data"]
+
+        # Is it necessary to create a cart in this case?
+        # This is done for backward compatibility for now
+        self.assertNotEqual(anon_cart.id, cart["id"])
```

Comment:
Isn't this test missing (restrict_cart_to_partner)?  self..assertEqual(anon_cart.partner_id, self.partner)